ToDo App - Jasmin H

JasminHed opened the pull request:
Link netlify: https://todoapp-jh.netlify.app/

JasminHed committed:
added global zustand store + adding todos function + 3 buttons with tasks

JasminHed committed:
added list + complete/uncomplete

JasminHed committed:
added remove+count+emptystate

JasminHed committed:
styling on mobile

JasminHed committed:
mediaq for tablet and desktop+scroll on form + new backg image

JasminHed committed:
added mediaq desktop, pink to checkboxes

JasminHed committed:
localstorage store.jsx and boxes same spacing

JasminHed committed:
added landmarks header+main

JasminHed committed:
added a animation postit

JasminHed committed:
updated animation, slower + added comments to code

JasminHed committed:
updated height on global, added tags, removed postit, height on form + maincontainer to auto

JasminHed committed:
added grid layout on tabelt + desktop, excel style, removed backgrondimage, removed boxes with gradient

JasminHed committed:
added backgroundimage back + backgroundcolor to form box

JasminHed committed:
updated w. comments, better styling on small screen mobile first

JasminHed committed:
added arialabel on checkbox + accent background on checkbox

JasminHed committed:
updated readmefile + added border and checked color on checkbox

JasminHed committed:
added character count

JasminHed committed:
styled charcount on tablet+desktop

HIPPIEKICK reviewed (changes requested):
Good that you’re limiting the input!! However, I found a couple of bugs that need fixing.

**Have a look at what happens when the user writes in the task input field - the character limit text jumps a bit. Another bug I found is if you’re adding a task without a tag and note, the checkbox shows up under the Notes, so your “columns” for the different things doesn’t really work.**

And I’d suggest you have a clearer visual indicator when the checkbox is checked because changing only the background doesn’t make it so clear. This is more of a nice-to-have and a personal opinion 😊

Don’t forget to be consistent in your code, e.g. use only arrow functions.

ToDo.jsx is quite a big file. Why not split it up? A low-hanging fruit is breaking out the header at least.

**Changes requested**
The bugs marked in bold

JasminHed committed:
eslint on indentation

JasminHed committed:
removedbug-new styling for input field+character count, removed bug-each column takes up space wether content or not

JasminHed committed:
added an after check styling for checkboxes

JasminHed committed:
changed function app to arrow function for consistency

JasminHed committed:
extracted todoitem , header + todoform from todo.jsx

JasminHed committed:
required tag + error message, column on 320-667px -no notes no header

JasminHed committed:
removed old code for todoitem container, checked mediaq, styling for column todoitemcontainer

JasminHed commented:
Hello Matilda,
Okay so this project, wanted so much made so little work :) 
- character limit text should now not jump at all
- bug shift in layout; added column layout w. flexbox for 320-667px, so that everything fits that need fitting. Also added tag as requirement and an error message to go with that. It should now function on all screens.
- checked so all my functions are arrow functions and changed accordingly
- clear visual indiciator on checkbox, great tip!
- went a bit crazy with making the todo.jsx cleaner, so now we have todoform, header and todoitem all in their own file. Also changed the styling to 1 box form + 1 box content of form. 

Hope this works better now!

HIPPIEKICK reviewed (approved):
Much better, I'm approving now! Just make sure to also fix the jumping when a user clicks a tag, see video 👇

https://github.com/user-attachments/assets/a4eff7c6-849b-4925-8b3c-a4485ead4019

JasminHed committed:
added transparent border on tagbutton to stop shift on page

JasminHed commented:
Aha! I added a border, transparent to the tag button and that actually made it stay put. Taking note for next time. Thank you!

JasminHed committed:
added enter handle keydown

JasminHed committed:
updated readmefile

JasminHed committed:
updated ackground color to neutral

JasminHed committed:
updated font-size for character count
